### PR TITLE
test: add host requirement integ tests

### PR DESCRIPTION
**Pull request by edwards-aws**
### What was the problem/requirement? (What/Why)
We need integ tests for host requirements in the job bundle!

I also noticed a new issue since the output paths are sticky now. Subsequent runs of a test would fail because the output path would be pulled from the sticky setting instead of the scene like it's supposed to be.

### What was the solution? (How)
Add the tests! I also did a tiny bit of refactoring to reduce duplicate code.

To fix the sticky setting issue I added a way to mark the tests with the scene files they use. All submitter tests will automatically remove the sticky settings file before they start if the sticky settings exist.

### What is the impact of this change?
More test coverage and confidence in our integrations :)

### How was this change tested?
`hatch run integ:test`

#### Please run the integration tests and paste the results below
```
================================================================== test session starts ==================================================================
platform darwin -- Python 3.12.7, pytest-8.3.4, pluggy-1.5.0 -- /Users/edwards/Library/Application Support/hatch/env/virtual/deadline-cloud-for-blender/ZLHXUBw2/integ/bin/python
cachedir: .pytest_cache
rootdir: /Users/edwards/git/deadline-cloud-for-blender
configfile: pyproject.toml
plugins: cov-6.0.0, xdist-3.6.1
1 worker [4 items]     
scheduling tests via LoadScheduling

test/integ/test_blender.py::TestSubmitters::test_minimal_scene_submitter 
[gw0] [ 25%] PASSED test/integ/test_blender.py::TestSubmitters::test_minimal_scene_submitter 
test/integ/test_blender.py::TestSubmitters::test_minimal_scene_with_host_requirements_submitter 
[gw0] [ 50%] PASSED test/integ/test_blender.py::TestSubmitters::test_minimal_scene_with_host_requirements_submitter 
test/integ/test_blender.py::TestAdaptors::test_minimal_scene_adaptor 
[gw0] [ 75%] PASSED test/integ/test_blender.py::TestAdaptors::test_minimal_scene_adaptor 
test/integ/test_blender.py::TestAdaptors::test_minimal_scene_with_host_requirements_adaptor 
[gw0] [100%] PASSED test/integ/test_blender.py::TestAdaptors::test_minimal_scene_with_host_requirements_adaptor 

================================================================== slowest 5 durations ==================================================================
26.00s call     test/integ/test_blender.py::TestAdaptors::test_minimal_scene_adaptor
12.50s call     test/integ/test_blender.py::TestAdaptors::test_minimal_scene_with_host_requirements_adaptor
1.35s call     test/integ/test_blender.py::TestSubmitters::test_minimal_scene_submitter
1.31s call     test/integ/test_blender.py::TestSubmitters::test_minimal_scene_with_host_requirements_submitter
0.00s setup    test/integ/test_blender.py::TestSubmitters::test_minimal_scene_submitter
================================================================== 4 passed in 41.43s ===================================================================
```

### Was this change documented?
Doesn't need to be

### Is this a breaking change?
Nope

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: test/integ/test_blender.py [hunk 1]
```diff
@@ -6,119 +6,258 @@
...
-        "ResolutionX": 640,
-        "ResolutionY": 480,
-    }
-
-    run_adaptor_test(test_file_location / "expected_job_bundle" / "template.yaml", job_params)
-    assert_all_images_close(
-        expected_image_directory=test_file_location / "expected_images",
-        actual_image_directory=output_path,
-    )
+class TestAdaptors:
```
Comment (leongdl) [inline test/integ/test_blender.py]: nit: What if this was a separate file ? More for long term maintainability as we add more tests.

Comment (edwards-aws) [reply test/integ/test_blender.py]: Makes sense to me!

Comment (edwards-aws) [reply test/integ/test_blender.py]: Changed!

**Review by leongdl — APPROVED** [review]
Thank you!

**Review by miabatta — APPROVED** [review]
Looks good! Tests pass with on with 4.2 on Windows.
